Author Books Endpoint

SammyT13 opened the pull request:
Created list of books by author's name endpoint.

SammyT13 committed:
Added Book endpoint.  Added Author endpoint.

SammyT13 committed:
Added Book endpoint.  Added Author endpoint.

SammyT13 committed:
Added Book endpoint.  Added Author endpoint.

SammyT13 committed:
Created author endpoint.

SammyT13 committed:
Merge remote-tracking branch 'upstream/master'

SammyT13 committed:
Created author endpoint.

Sullivan-Patrick commented on src/main/java/com/cen4010/bookstore/book/Book.java:
```diff
@@ -3,20 +3,20 @@
 
 
 @Entity
+@Table
```
What's the purpose of this annotation? For that matter, what's the purpose of making any modifications to the book entity? It's been working fine.

Sullivan-Patrick commented on src/main/java/com/cen4010/bookstore/book/Book.java:
```diff
@@ -3,20 +3,20 @@
 public class Book {
 
   @Id
+  @GeneratedValue
```
What's the purpose of this annotation? For that matter, what's the purpose of making any modifications to the book entity? It's been working fine.

SammyT13 replied:
So if I don't have @GeneratedValue my HTTP request all come back with 500 response.  The message would say "ids for this class must be manually assigned before calling save()" I found this to be an easy fix for it. I didn't know if this would impact anyone else's code.

Sullivan-Patrick replied:
Were you able to figure out why there was a 500 response? There should be an associated stacktrace. I believe what you're doing is making it auto-assign an id, but we should already have an id. From my own experience here, Axel added a UUID but we actually have a long which acts as the real PK of the table. @Axel-Soto can you weigh in?

SammyT13 replied:
I looked at Axel-Soto's code.  For credit card entity he has @Generated something and for his user entity he's using UUID but he also has a dto for that.

Sullivan-Patrick commented on src/main/java/com/cen4010/bookstore/book/Book.java:
```diff
@@ -69,49 +73,87 @@ public Book(UUID id, String ISBN, String name, String description, double price,
     this.copiesSold = copiesSold;
   }
 
-  public Book() {
```
Why remove default constructor? IIRC it may be necessary for some functionalities of Spring JPA

SammyT13 replied:
So I actually moved the public Book() constructor to the top before the other two constructors.  I guess my OCD kicked in just thought it was better on top.

Sullivan-Patrick replied:
My bad, just misread the code. No problem here.

Sullivan-Patrick commented on src/main/java/com/cen4010/bookstore/book/BookRepository.java:
```diff
@@ -1,14 +1,24 @@
-  List<Book> findByGenre(String genre);
 
-  List<Book> findTop10ByOrderByCopiesSoldDesc();
+
```
Why are you removing `findBy` methods others have explicitly added? Have you tested to make sure you're not breaking others code?

SammyT13 replied:
I didn't change that.  It's still in there.  I'm looking at the BookService on my local and your repo it's still the same.  I'm not quite sure the reference.

Sullivan-Patrick replied:
GitHub diff stuff- again, my bad.

Sullivan-Patrick commented on src/main/resources/application.properties:
```diff
@@ -1,9 +1,19 @@
-spring.jpa.show-sql=true
-spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.PostgreSQLDialect
-spring.jpa.properties.hibernate.format_sql=true
+#spring.datasource.url=jdbc:postgresql://localhost:5432/postgres
```
Please don't change this file on origin. See the comment above

SammyT13 replied:
I know I added my own application properties for my database, but I did comment them out so yes you can see them but it wouldn't pull anything.  I guess I was being lazy about it, I can copy paste them next time and un-comment the original.  I wasn't really trying to figure out gitignore at this moment either.

Sullivan-Patrick replied:
just don't `git add` the file and the commit won't alter the file

SammyT13 committed:
Added findBookByIsbn in BookService.
Updated addBook (changed name) in BookService.
Added @GetMapping to getBookyByIsbn in BookController.
Changed @GetMapping path for getAuthor in BookController.

SammyT13 committed:
Forgot to remove a parameter from getBook in BookController.  I wasn't trying to change it.  I was playing around with it a little.

SammyT13 committed:
Merge remote-tracking branch 'upstream/master'

SammyT13 committed:
Fixed the missing "confirmBookByIsbn."  I'm still using @GeneratedValue.  If I don't use it I get an error when trying to POST.  The error states an ID would need to manually created.

SammyT13 committed:
Created Author Entity.  This allows to create an author endpoint only.

Sullivan-Patrick reviewed (approved):
looks ok, keeping in mind we will need to revert if this breaks existing endpoints.
